s4e-backend/src/main/java/pl/cyfronet/s4e/data/repository/InvitationRepository.java

```diff
@@ -0,0 +1,16 @@
+package pl.cyfronet.s4e.data.repository;
+
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.transaction.annotation.Transactional;
+import pl.cyfronet.s4e.bean.Invitation;
+
+import java.util.Optional;
+import java.util.Set;
+
+@Transactional(readOnly = true)
+public interface InvitationRepository extends CrudRepository<Invitation, Long> {
+    <T> Optional<T> findByEmailAndInstitutionSlug(String email, String institutionSlug, Class<T> projection);
+    <T> Optional<T> findByToken(String token, Class<T> projection);
+    <T> Optional<T> findById(Long id, Class<T> projection);
+    <T> Set<T> findAllByInstitution(String institutionSlug, Class<T> projection);
```

Comment:
This won't work as you have a type mismatch between Institution defined in the method name and instSlug in an argument. However, as you don't use this method (the service method it uses is unused) I'd just remove it altogether.